src/Fixer/Basic/BracesFixer.php

```diff
@@ -354,13 +354,35 @@ function ($item) {
                     continue;
                 }
 
-                if (1 === $nestLevel && $nestToken->equalsAny(array(';', '}'))) {
+                $nextLineCanBeIndented = false;
+                if (1 === $nestLevel) {
+                    if ($nestToken->equalsAny(array(';', '}'))) {
+                        $nextLineCanBeIndented = true;
+                    } elseif ($nestToken->isComment()) {
+                        for ($i = $nestIndex - 1; $i > $startBraceIndex; --$i) {
+                            if ($tokens[$i]->isWhitespace()) {
+                                $nextLineCanBeIndented = (bool) preg_match('/\R/', $tokens[$i]->getContent());
+
+                                break;
+                            }
+                        }
+
+                        if ($i === $startBraceIndex) {
+                            $nextLineCanBeIndented = true;
+                        }
+                    }
+                }
+
+                if ($nextLineCanBeIndented) {
```

Comment:
this check can be moved at the end inside the block `if (1 === $nextlevel)` (i.e. remove line `}` at ln 374 and close later)